internal/pkg/telemetry/metric/prometheus/prometheus_test.go

```diff
@@ -27,10 +27,8 @@ func TestServeMetrics(t *testing.T) {
 
 	d := dependencies.NewMocked(t)
 
-	port, err := netutils.FreePort()
-	assert.NoError(t, err)
-
 	// Serve metrics
+	port := netutils.FreePortForTest(t)
 	listenAddr := fmt.Sprintf("localhost:%d", port)
```

Comment:
I improved it by adding a FS lock, we'll see if it will be enough or if it will crash again.

Port collisions were rare, it should be enough.